topic_tools/CMakeLists.txt

```
@@ -85,8 +85,63 @@ target_link_libraries(mux
 )
 
 if(BUILD_TESTING)
+  find_package(std_msgs REQUIRED)
+
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
+
+  find_package(ament_cmake_gtest REQUIRED)
+  ament_add_gtest(test_relay test/test_relay.cpp)
+  target_include_directories(test_relay PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+  )
+  ament_target_dependencies(test_relay
+    rclcpp
+    std_msgs
+  )
+  target_link_libraries(test_relay
+    relay_node
+  )
```

Nit: I've been convinced that we should prefer using modern CMake targets, moving away from `ament_target_dependencies` wherever possible (the one part I don't really like is that you can `lib::lib` for most things but need `${pkg_TARGETS}` for interface libraries)

```suggestion
  target_link_libraries(test_relay
    relay_node
    rclcpp::rclcpp
    ${std_msgs_TARGETS}
  )
```

This would apply to all your new test targets

---

Done.